backends/source/dppl_sycl_queue_interface.cpp

```diff
@@ -34,32 +34,230 @@ using namespace cl::sycl;
 namespace
 {
 // Create wrappers for C Binding types (see CBindingWrapping.h).
- DEFINE_SIMPLE_CONVERSION_FUNCTIONS(queue, DPPLSyclQueueRef)
+DEFINE_SIMPLE_CONVERSION_FUNCTIONS(context, DPPLSyclContextRef)
+DEFINE_SIMPLE_CONVERSION_FUNCTIONS(device, DPPLSyclDeviceRef)
+DEFINE_SIMPLE_CONVERSION_FUNCTIONS(event, DPPLSyclEventRef)
+DEFINE_SIMPLE_CONVERSION_FUNCTIONS(kernel, DPPLSyclKernelRef)
+DEFINE_SIMPLE_CONVERSION_FUNCTIONS(queue, DPPLSyclQueueRef)
+
+/*!
+ * @brief Set the kernel arg object
+ *
+ * @param    cgh            My Param doc
+ * @param    Arg            My Param doc
+ */
+bool set_kernel_arg (handler &cgh, size_t idx, __dppl_keep void *Arg,
```

Comment:
We will eventually need to support complexes as well. 

This is not straightforward with `ctypes`: https://stackoverflow.com/questions/13373291/complex-number-in-ctypes

So down the road we might just introduce our own scalar type in `dpctl`.

Reply:
Opened #85 to track the issue.